Plugins/Animation/src/AnimationComponent.hpp

```diff
@@ -18,39 +18,87 @@ namespace AnimationPlugin {
 
 class SkeletonBoneRenderObject;
 
+/// The AnimationComponent is responsible for the management of the animations of an Entity.
+/// It stores the animation Skeleton and the animation data as well as the animation playing speed.
 class ANIM_PLUGIN_API AnimationComponent : public Ra::Engine::Component {
-  public:
+public:
     AnimationComponent( const std::string& name, Ra::Engine::Entity* entity );
     virtual ~AnimationComponent();
     AnimationComponent( const AnimationComponent& ) = delete;
     AnimationComponent& operator=( const AnimationComponent& ) = delete;
 
     virtual void initialize() override {}
 
+    //
+    // Build from fileData
+    //
+
+    /// Create the skeleton from the given data.
+    /// @param data: the skeleton's joint transform hierarchy.
+    /// @param duplicateTable: the map from duplicate to single vertices.
+    /// @param the number of single vertices.
+    // FIXME: the 2 last parameters are needed only for the genereation of
+    //        the skinning weights matrix.
+    void handleSkeletonLoading( const Ra::Asset::HandleData* data,
+                                const std::vector<Ra::Core::Index>& duplicateTable,
+                                uint nbMeshVertices );
```

Comment:
It is not clear to me why we still have the duplicate table.
Isn't the attribute system supposed to fix this ?

Reply:
Well, for now, the skinnnig weights are not stored as attributes but as a matrix.
And when creating this matrix, which has a row per duplicated position, we must know how to map the loaded indices (in GeometryData) to the stored indices (in TriangleMesh).

Reply:
Please fix this in a dedicated PR.